src/opm/output/eclipse/AggregateWellData.cpp

```diff
@@ -517,17 +517,17 @@ namespace {
                 const auto& pc = well.productionControls(smry);
                 const auto& predMode = well.predictionMode();
 
-                if (pc.oil_rate != 0.0) {
+                if ((pc.oil_rate != 0.0) || (!predMode)) {
```

Comment:
I think this would be easier to read if the `predMode` test was on the "outside"; i.e. something like:
```C++
if (!predMode) {
     if (pc.oil_rate != 0) 
          sWell[] = 

    if (pc.water_rate != 0)
          sWell[] = 

    if (pc.gas_rate != 0)
          sWell[] = 
}
```

Reply:
I have restructured the test as suggested to make it easier to read. However, the content inside the test is not as suggested in the comment, which probably shows the need to rewrite!